### PR TITLE
Remove duplicated group_users before saving the group

### PR DESCRIPTION
**Pull request by allenwq**
Also removed current users from association list when create the group

### CODE REVIEW

File: app/models/course/group.rb [hunk 1]
```diff
@@ -20,4 +21,12 @@ def set_defaults
     return if !course.course_users.exists?(user: creator) || group_users.exists?(user: creator)
     group_users.build(user: creator, role: :manager, creator: creator, updater: updater)
   end
+
+  def remove_duplicated_group_users
+    new_group_users = group_users.select(&:new_record?)
```
Comment (lowjoel) [inline app/models/course/group.rb]: This is an O(n^2) algorithm. Please make it linear.

File: app/models/course/group.rb [hunk 2]
```diff
@@ -20,4 +21,12 @@ def set_defaults
     return if !course.course_users.exists?(user: creator) || group_users.exists?(user: creator)
     group_users.build(user: creator, role: :manager, creator: creator, updater: updater)
   end
+
+  def remove_duplicated_group_users
```
Comment (lowjoel) [inline app/models/course/group.rb]: `remove_duplicate_users`?

File: app/models/course/group.rb [hunk 3]
```diff
@@ -20,4 +21,16 @@ def set_defaults
     return if !course.course_users.exists?(user: creator) || group_users.exists?(user: creator)
     group_users.build(user: creator, role: :manager, creator: creator, updater: updater)
   end
+
+  def remove_duplicate_users
```
Comment (allenwq) [inline app/models/course/group.rb]: @lowjoel  I've updated the algorithm,  think it's linear now

Comment (lowjoel) [inline app/models/course/group.rb]: still quadratic. Use a set and query whether an object is in the set. `count` is still linear.

Comment (lowjoel) [inline app/models/course/group.rb]: ...can you just use `uniq` with a block?

Comment (allenwq) [inline app/models/course/group.rb]: I can only use `uniq` on new records( I don't want persisted records been affected), have updated the code, is it better now ?

**Pull request merged**